host/sgx/enter.c

```diff
@@ -181,8 +181,7 @@ void oe_enter_sim(
     OE_ALIGNED(16)
     uint64_t fx_state[64];
 
-    // Backup host GS and FS registers.
-    void* host_gs = oe_get_gs_register_base();
```

Comment:
Can you also clean up existing references to `oe_get_gs_register_base()` and possibly remove the method entirely if unused? It only seems to be called in a vestigial codepath in `GetThreadBinding()` when `USE_TLS_FOR_THREADING_BINDING`, which is always defined and could also probably be removed as a meaningless macro.

The comments in this file also still refer to storing thread data in the GS segment register, which is confusing, since it looks like it's all TLS-based now. 

On a related note, I'm not sure how `GetThreadBinding()` and `ThreadBinding` escaped the variable rename change to `oe_get_thread_binding()` and `oe_thread_binding_t` respectively, but we should probably track fixing that up as well if not addressed in this PR.

Reply:
Good point. I have cleaned up the naming convention and comments,  and also removed `USE_TLS_FOR_THREAD_BINDING`. They don't add much to the bulk of the PR and are within its scope.